synapse/api/room_versions.py

```diff
@@ -259,6 +271,24 @@ class RoomVersions:
         msc2403_knocking=True,
         msc2716_historical=True,
         msc2716_redactions=True,
+        msc2716_empty_prev_events=False,
+    )
+    MSC2716v4 = RoomVersion(
```

Comment:
New room version split this out to https://github.com/matrix-org/synapse/pull/11243 and will disappear in the diff here after it's merged.